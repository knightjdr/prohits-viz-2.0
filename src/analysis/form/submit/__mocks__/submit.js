import React from 'react';

const Submit = () => (
  <div className="Submit" />
);
export default Submit;
