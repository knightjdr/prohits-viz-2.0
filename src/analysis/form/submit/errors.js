import React from 'react';

import './errors.css';

const Errors = (errors) => {
  if (Object.keys(errors).length > 0) {
    const content = Object.entries(errors).map(([key, error]) => (
      <div key={key}>{error}</div>
    ));
    return (
      <div className="Errors-container">
        <div className="Errors-header">
          Submission errors
        </div>
        <div className="Errors-content">
          {content}
        </div>
      </div>
    );
  }
  return null;
};
export default Errors;
