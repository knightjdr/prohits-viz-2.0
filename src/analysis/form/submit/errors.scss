.Errors-container {
  background-color: rgba(211, 47, 47, 0.2);
  border: 1px solid rgba(211, 47, 47, 0.7);
  border-radius: 5px;
  margin-top: 25px;
  padding: 10px;
}
.Errors-content {
  font-size: 0.9em;
}
.Errors-header {
  font-weight: bold;
}
