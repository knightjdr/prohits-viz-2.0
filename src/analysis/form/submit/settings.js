import React from 'react';
import { Tag } from 'antd';

import FalsyButNotZero from '../../../helpers/falsy-but-not-zero';

const validTypes = {
  logBase: [2, 'e', 10],
  normalization: ['readout', 'total'],
};

const Settings = (form) => {
  const tagArray = [];
  switch (form.analysisType) {
    case 'dotplot':
      tagArray.push({
        key: 'primaryFilter',
        text: `Primary filter: ${FalsyButNotZero(form.primaryFilter) ? 'not set' : form.primaryFilter}`,
      });
      tagArray.push({
        key: 'secondaryFilter',
        text: `Secondary filter: ${FalsyButNotZero(form.secondaryFilter) ? 'not set' : form.secondaryFilter}`,
      });
      tagArray.push({
        key: 'minAbundance',
        text: `Minimum abundance: ${FalsyButNotZero(form.minAbundance) ? 'not set' : form.minAbundance}`,
      });
      if (form.ctrlSub) {
        tagArray.push({
          key: 'ctrlSub',
          text: 'Control subtraction is selected',
        });
      }
      if (form.readoutLengthNorm) {
        tagArray.push({
          key: 'readoutLengthNorm',
          text: 'Readout length normalization is selected',
        });
      }
      if (
        form.normalization &&
        validTypes.normalization.includes(form.normalization)
      ) {
        tagArray.push({
          key: 'normalization',
          text: `Condition normalization: ${form.normalization === 'total' ? 'total abundance' : 'specific readout'}`,
        });
      }
      if (
        form.logBase &&
        validTypes.logBase.includes(form.logBase)
      ) {
        tagArray.push({
          key: 'logBase',
          text: `Log transformation: base ${form.logBase}`,
        });
      }
      tagArray.push({
        key: 'clustering',
        text: `Clustering type: ${form.clustering || 'not set'}`,
      });
      return (tagArray.map(tag => (
        <Tag
          className="submit__tag"
          key={tag.key}
        >
          {tag.text}
        </Tag>
      )));
    default:
      return null;
  }
};
export default Settings;
