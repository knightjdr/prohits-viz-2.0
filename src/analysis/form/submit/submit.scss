@import 'style/button';

.submit {
  margin-top: 20px;
}
.submit__buttons {
  align-items: center;
  display: grid;
  grid-template-columns: auto auto;
  grid-template-rows: auto auto auto;
  margin-top: 27px;
}
.submit__buttons > *:not(:last-child) {
  margin-bottom: 10px;
}
.submit__buttons-options {
  grid-column: 1;
  grid-row: 3;
}
.submit__buttons-submit {
  @extend .success-button;
  grid-column: 1;
  grid-row: 1;
}
.submit__buttons-warning {
  @extend .warning-button;
  grid-column: 1;
  grid-row: 2;
}
.submit_pending {
  grid-column: 2;
  grid-row: 1;
}
.submit_pending > :first-child{
  margin: 0 5px;
}
.submit__settings {
  display: flex;
  flex-direction: column;
  margin-right: 20px;
}
.submit__settings > *:not(:last-child) {
  margin-bottom: 5px;
}
.submit__status {
  align-items: flex-start;
  display: flex;
  flex-wrap: wrap;
  margin-top: 10px;
}
.submit__tag {
  cursor: default;
}

