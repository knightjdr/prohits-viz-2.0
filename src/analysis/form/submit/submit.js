import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';
import React from 'react';
import { Button } from 'antd';
import { connect } from 'react-redux';
import {
  faMinusSquare,
  faPlusSquare,
  faSpinner,
} from '@fortawesome/pro-solid-svg-icons';

import AnalysisFormSelector from '../../../state/selectors/analysis-form-selector';
import Errors from './errors';
import Settings from './settings';

import './submit.css';

export const SubmitComponent = ({
  errors,
  form,
  handleOptions,
  handleReset,
  showOptions,
  submitPending,
}) => {
  const optionsIcon = showOptions ?
    (
      <FontAwesomeIcon
        className="submit__hide-options-icon"
        icon={faMinusSquare}
        size="sm"
      />
    )
    :
    (
      <FontAwesomeIcon
        className="submit__show-options-icon"
        icon={faPlusSquare}
        size="sm"
      />
    );
  const headerElement = (
    <div className="submit">
      <div>
        Hit the submit button when ready or customize options.
      </div>
      <div className="submit__status">
        <div className="submit__settings">
          <div>
            Current settings:
          </div>
          { Settings(form) }
        </div>
        <div className="submit__buttons">
          <Button
            className="submit__buttons-submit"
            disabled={submitPending}
            htmlType="submit"
            type="submit"
          >
            Submit
          </Button>
          {
            submitPending &&
            <span className="submit_pending">
              <FontAwesomeIcon icon={faSpinner} spin />
              submission pending
            </span>
          }
          <Button
            className="submit__buttons-warning"
            onClick={handleReset}
            tooltip="Reset form"
            tooltip-placement="left"
            type="button"
          >
            Reset
          </Button>
          <Button
            className="submit__buttons-options"
            onClick={handleOptions}
            type="primary"
          >
            {optionsIcon} Options
          </Button>
        </div>
        {Errors(errors)}
      </div>
    </div>
  );
  return (
    headerElement
  );
};

SubmitComponent.propTypes = {
  errors: PropTypes.shape({}).isRequired,
  form: PropTypes.shape({}).isRequired,
  handleOptions: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
  showOptions: PropTypes.bool.isRequired,
  submitPending: PropTypes.bool.isRequired,
};

/* istanbul ignore next */
const mapStateToProps = state => ({
  form: AnalysisFormSelector(state),
});

const ConnectedContainer = connect(
  mapStateToProps,
)(SubmitComponent);

export default ConnectedContainer;
