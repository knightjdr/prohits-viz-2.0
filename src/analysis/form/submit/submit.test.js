import React from 'react';
import { shallow } from 'enzyme';

import { SubmitComponent } from './submit';

jest.mock('./errors');
jest.mock('./settings');

describe('Submit component', () => {
  describe('default view', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <SubmitComponent
          errors={{}}
          form={{}}
          handleOptions={jest.fn()}
          handleReset={jest.fn()}
          showOptions={false}
          submitPending={false}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not show options', () => {
      expect(wrapper.find('.submit__show-options-icon').length).toBe(1);
    });

    it('should not show pending element', () => {
      expect(wrapper.find('.submit_pending').length).toBe(0);
    });
  });

  describe('showing options', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <SubmitComponent
          errors={{}}
          form={{}}
          handleOptions={jest.fn()}
          handleReset={jest.fn()}
          showOptions
          submitPending={false}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not show options', () => {
      expect(wrapper.find('.submit__hide-options-icon').length).toBe(1);
    });

    it('should not show pending element', () => {
      expect(wrapper.find('.submit_pending').length).toBe(0);
    });
  });

  describe('when pending', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <SubmitComponent
          errors={{}}
          form={{}}
          handleOptions={jest.fn()}
          handleReset={jest.fn()}
          showOptions={false}
          submitPending
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should show pending element', () => {
      expect(wrapper.find('.submit_pending').length).toBe(1);
    });
  });
});
