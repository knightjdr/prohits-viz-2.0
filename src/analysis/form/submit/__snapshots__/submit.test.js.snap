// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit component default view should match snapshot 1`] = `
<div
  className="submit"
>
  <div>
    Hit the submit button when ready or customize options.
  </div>
  <div
    className="submit__status"
  >
    <div
      className="submit__settings"
    >
      <div>
        Current settings:
      </div>
    </div>
    <div
      className="submit__buttons"
    >
      <Button
        className="submit__buttons-submit"
        disabled={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        prefixCls="ant-btn"
        type="submit"
      >
        Submit
      </Button>
      <Button
        className="submit__buttons-warning"
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        tooltip="Reset form"
        tooltip-placement="left"
        type="button"
      >
        Reset
      </Button>
      <Button
        className="submit__buttons-options"
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        type="primary"
      >
        <FontAwesomeIcon$1
          border={false}
          className="submit__show-options-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f0fe",
                "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
              ],
              "iconName": "plus-square",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size="sm"
          spin={false}
          symbol={false}
          transform={null}
        />
         Options
      </Button>
    </div>
  </div>
</div>
`;

exports[`Submit component showing options should match snapshot 1`] = `
<div
  className="submit"
>
  <div>
    Hit the submit button when ready or customize options.
  </div>
  <div
    className="submit__status"
  >
    <div
      className="submit__settings"
    >
      <div>
        Current settings:
      </div>
    </div>
    <div
      className="submit__buttons"
    >
      <Button
        className="submit__buttons-submit"
        disabled={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        prefixCls="ant-btn"
        type="submit"
      >
        Submit
      </Button>
      <Button
        className="submit__buttons-warning"
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        tooltip="Reset form"
        tooltip-placement="left"
        type="button"
      >
        Reset
      </Button>
      <Button
        className="submit__buttons-options"
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        type="primary"
      >
        <FontAwesomeIcon$1
          border={false}
          className="submit__hide-options-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f146",
                "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z",
              ],
              "iconName": "minus-square",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size="sm"
          spin={false}
          symbol={false}
          transform={null}
        />
         Options
      </Button>
    </div>
  </div>
</div>
`;

exports[`Submit component when pending should match snapshot 1`] = `
<div
  className="submit"
>
  <div>
    Hit the submit button when ready or customize options.
  </div>
  <div
    className="submit__status"
  >
    <div
      className="submit__settings"
    >
      <div>
        Current settings:
      </div>
    </div>
    <div
      className="submit__buttons"
    >
      <Button
        className="submit__buttons-submit"
        disabled={true}
        ghost={false}
        htmlType="submit"
        loading={false}
        prefixCls="ant-btn"
        type="submit"
      >
        Submit
      </Button>
      <span
        className="submit_pending"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f110",
                "M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z",
              ],
              "iconName": "spinner",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={true}
          symbol={false}
          transform={null}
        />
        submission pending
      </span>
      <Button
        className="submit__buttons-warning"
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        tooltip="Reset form"
        tooltip-placement="left"
        type="button"
      >
        Reset
      </Button>
      <Button
        className="submit__buttons-options"
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        type="primary"
      >
        <FontAwesomeIcon$1
          border={false}
          className="submit__show-options-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f0fe",
                "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
              ],
              "iconName": "plus-square",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size="sm"
          spin={false}
          symbol={false}
          transform={null}
        />
         Options
      </Button>
    </div>
  </div>
</div>
`;
